src/qibo/tests/test_abstract_circuit.py

```diff
@@ -117,6 +117,19 @@ def gen():
     assert isinstance(c.queue[5], gates.CNOT)
     assert isinstance(c.queue[7], gates.H)
 
+def test_circuit_add_repeated_execution():
+    c = Circuit(1)
+    c.add(gates.H(0))
+    c.add(gates.PauliNoiseChannel(0, px=0.2))
+
+    c1 = Circuit(1)
+    c1.add(gates.X(0))
+
+    c2 = c + c1
+
+    assert c.repeated_execution
+    assert not c1.repeated_execution
+    assert c2.repeated_execution
```

Comment:
Looks okay to me, however maybe one should include an example where you compare the shots directly (after fixing the seed).

Reply:
I agree with adding a test using the example from #563. You can add this to one of the `test_core_circuit_*.py` files (for example `test_core_circuit_noise.py`) because `test_abstract_circuit.py` is not supposed to test execution and is backend free.

Reply:
Sure, I also agree with adding the test to compare shots.
At this point should I remove this test?

Reply:
>  At this point should I remove this test?

I would say yes.

Reply:
Done.